### PR TITLE
NumberBox: float values with percent format should be rounded correctly (T1093736)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 1]
```diff
@@ -1330,6 +1330,25 @@ QUnit.module('format: percent format', moduleConfig, () => {
...
+        { text: '0.04', value: 0.00035, format: '#0.00%' },
+        { text: '0.0350', value: 0.00035, format: '#0.0000%' },
+        { text: '0.14', value: 0.00135, format: '#0.00%' },
+        { text: '0.4', value: 0.0035, format: '#0.0%' },
+        { text: '0.35', value: 0.0035, format: '#0.00%' },
+        { text: '1.4', value: 0.0135, format: '#0.0%' },
+        { text: '0.005', value: 0.000049999, format: '#0.000%' },
+        { text: '0.004', value: 0.0000444999, format: '#0.000%' },
+    ].forEach(({ text, value, format }) => {
+        QUnit.test(`percent format should correctly handle float values, value: ${value}, format: ${format}`, function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]:
```suggestion
        QUnit.test(`percent format should correctly handle float values, value: ${value}, format: ${format} (T1093736)`, function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 2]
```diff
@@ -1330,6 +1330,25 @@ QUnit.module('format: percent format', moduleConfig, () => {
...
+            this.instance.option('format', format);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]:
```suggestion
            this.instance.option({
                format,
                value
            });
```

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 3]
```diff
@@ -1330,6 +1330,25 @@ QUnit.module('format: percent format', moduleConfig, () => {
...
+            this.instance.option('value', value);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]:
```suggestion
```

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 4]
```diff
@@ -1330,6 +1330,25 @@ QUnit.module('format: percent format', moduleConfig, () => {
...
+
+            assert.equal(this.input.val(), `${text}%`, 'text is correct');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]:
```suggestion
            assert.strictEqual(this.input.val(), `${text}%`, 'text is correct');
```

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 5]
```diff
@@ -1330,6 +1330,25 @@ QUnit.module('format: percent format', moduleConfig, () => {
...
+            assert.equal(this.instance.option('value'), value, 'value is correct');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]:
```suggestion
            assert.strictEqual(this.instance.option('value'), value, 'value is correct');
```

**Comment by ksercs** [comment]
LGTM

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 5 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21833/commits/378a01f352b28c69e4f273bc35c12c312258ecec

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 4 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21833/commits/378a01f352b28c69e4f273bc35c12c312258ecec

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21833/commits/378a01f352b28c69e4f273bc35c12c312258ecec

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21833/commits/378a01f352b28c69e4f273bc35c12c312258ecec

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21833/commits/378a01f352b28c69e4f273bc35c12c312258ecec

**Review by Zedwag — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**